File: backend/app/db/mongodb.py
import motor.motor_asyncio
from decouple import config

MONGODB_URL = config("MONGODB_URL", default="mongodb://localhost:27017/english-tutor")

client = motor.motor_asyncio.AsyncIOMotorClient(MONGODB_URL)
db = client.get_database()


async def connect_to_mongo():
    try:
        await client.admin.command("ping")
        print("Successfully connected to MongoDB")
    except Exception as e:
        print(f"Error connecting to MongoDB: {e}")
        raise e

File: frontend/lib/mongodb.ts
import { MongoClient, ServerApiVersion } from "mongodb";

const uri = process.env.MONGODB_URI
if (!uri) {
  throw new Error("MONGODB_URI is not set");
}


const client = new MongoClient(uri, {
  serverApi: {
    version: ServerApiVersion.v1,
    strict: true,
    deprecationErrors: true,
  },
});

async function connectDB() {
  try {
    if (!client.connect) return;
    await client.connect();
    await client.db("admin").command({ ping: 1 });
    console.log("Successfully connected to MongoDB.");
  } catch (error) {
    console.error("Error connecting to MongoDB:", error);
    throw error;
  }
}

export { connectDB };
export default client;
